### PR TITLE
the-way: init at 0.3.2

### PR DESCRIPTION
**Pull request by numkem**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/the-way/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "out-of-cheese-error";
+    repo = pname;
+    rev = version;
+    sha256 = "0vnqrlpmrc9mz1snd94w7rcxd5s7jfc84pl8j2hd01cvzfqjqjf3";
+  };
+
+  cargoSha256 = "0l5yanrlx0km82w7rpa9kq6sq9dnb40im0hddabdrgjc2934m19v";
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/tools/the-way/default.nix:15]: do you mind adding a comment as to why tests were disabled?
```suggestion
  doCheck = false;
```

File: pkgs/development/tools/the-way/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A terminal code snippets manager";
```
Comment (jonringer) [inline pkgs/development/tools/the-way/default.nix:18]: Personal preference to remove articles
```suggestion
    description = "Terminal code snippets manager";
```

**Comment by Gaelan** [comment]
Don't forget to add this to `all-packages.nix` :)

File: pkgs/development/tools/the-way/default.nix (hunk 1 above)
Comment (numkem) [reply pkgs/development/tools/the-way/default.nix:17]: Added it as the comment inside the derivation.

**Comment by numkem** [comment]
@jonringer I've fixed the tests so they are running correctly and updated the package to use version 0.3.2 instead of 0.2.3.

File: pkgs/development/tools/the-way/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "the-way";
+  version = "v0.3.2";
```
Comment (Lassulus) [inline pkgs/development/tools/the-way/default.nix:5]:
```suggestion
  version = "0.3.2";
```

File: pkgs/development/tools/the-way/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "out-of-cheese-error";
+    repo = pname;
+    rev = version;
```
Comment (Lassulus) [inline pkgs/development/tools/the-way/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/tools/the-way/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "out-of-cheese-error";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "02aa4iwwi89r6sj1q5sj74n2cy1rj94yfh39cp97zlx4lam9pj6b";
+  };
+
+  cargoSha256 = "1360ls1i5v0x4ksmrzhvh5mvpl1fx8zr8w3sp9rs8hqkrxa78qfa";
```
Comment (Lassulus) [inline pkgs/development/tools/the-way/default.nix:14]: seems the hash doesn't match: 
```
hash mismatch in fixed-output derivation '/nix/store/47mviv4z6s6dmwm2ph714sb9ghbbn1x6-the-way-0.3.2-vendor.tar.gz':
  wanted: sha256:1360ls1i5v0x4ksmrzhvh5mvpl1fx8zr8w3sp9rs8hqkrxa78qfa
  got:    sha256:09d9gaa83djfkfdx3m4d3kkq550q8y67wb292mrj4xb0kcfrhnpq
```

**Pull request merged**